### PR TITLE
Deleting disabled category forces category flat and catalog search re-indexed fully

### LINKED ISSUE #23297: Deleting disabled category forces category flat and catalog search re-indexed fully
### Description (*)
 When deleting a disabled category having no child and not product assigned to it then the status of Category Flat Data and Catalog search updated to Re-index Required 

### Steps to reproduce (*)

1. Create a disabled category
2. Run`indexer:reindex` to make sure that all indexer valid
3. Set all indexer to `Update by schedule` mode
4. Delete category (from step 1)
5. Observe indexers status

### Expected behavior (*)
 If a category is inactive then there should be no effect expected on catalog search and category flat index tables 

### Actual result (*)
Indexers have `invalid` status

### Benefits
 If this event happens Magento cron will try to full re-index and as indexing is time and resource consuming process then it may add some extra load unnecessarily. So we can avoid such things and save resource consumption if this event happens.

### Additional information
Magento version: 2.4-develop
Indexer Mode: Update By Schedule

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#23297
### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Integration Tests

**Comment by engcom-Charlie** [comment]
@magento run Integration Tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by danielrussob** [comment]
@magento give me 2.4-develop instance

**Comment by danielrussob** [comment]
@magento give me test instance

File: dev/tests/integration/testsuite/Magento/Catalog/Model/Indexer/Category/ProductTest.php [hunk 1]
```diff
@@ -184,7 +184,36 @@ public function testCategoryDelete()
...
+     *
+     * @magentoAppArea adminhtml
+     * @magentoDataFixture Magento/Catalog/_files/categories_disabled.php
+     *
+     * @return void
+     */
+    public function testDeleteInactiveCategory(): void
+    {
+        $this->indexer->reindexAll();
+        $indexerShouldBeValid = $this->indexer->isInvalid();
```
Comment (aleron75) [inline dev/tests/integration/testsuite/Magento/Catalog/Model/Indexer/Category/ProductTest.php]: Hello @engcom-Charlie thanks for your contribution!

The name of `$indexerShouldBeValid` variable wasn't very intuitive to me, it didn't give me the idea that we were getting the state of the indexer.

What about renaming with something like `$isInvalidIndexer`?

This way the assertion `$this->assertFalse($isInvalidIndexer)` becomes more readable too.

What do you think?

**Review by danielrussob — COMMENTED** [review]
Hi @engcom-Charlie , thank you for your contribution. I have a question for you!

I have tested this case:

Vanilla:
- Disable Category: Man (children are Enabled)
- Wait to reindex
- Delete Category: Man 
- See indexer status

image (1)

This PR:
- Disable Category: Man (children are Enabled)
- Wait to reindex
- Delete Category: Man 
- See indexer status

image (2)

I'm expecting to see a "Required Reindex" for children categories, why I can't see that condition?

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-8801](https://jira.corp.magento.com/browse/ENGCOM-8801) has been created to process this Pull Request

**Review by danielrussob — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @danielrussob, thank you for the review.
[ENGCOM-8801](https://jira.corp.magento.com/browse/ENGCOM-8801) has been created to process this Pull Request

**Comment by aleron75** [comment]
@engcom-Charlie can you please check whether the failing functional test is related to these PR changes?
Thanks

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario
1. From Admin Panel, go to _Catalog - Categories_
2. Click **Add Subcategory**
3. Type a _Category Name_
4. Set _Enable Category_ to **No**
5. **Save** category
6. Go to _System - Index Management_
7. Select All Indexers and **Submit** _Update by schedule_
8. Run `bin/magento indexer:reindex` and assert that all the indexers are in Status - Ready (green)
9. Go to _Catalog - Categories_ and **Delete** the previously created disabled Category
10. Go to _System - Index Management_ and verify the Indexers statuses 

**BEFORE** applying changes provided in the PR, some of the Indexers changed the Status to _Require Reindex_. Which is not correct as the deleted category was Disabled. And before deleting, full reindex has been done
before_indexer

**AFTER** switching to the PR, all the indexers are in Status - Ready (green)
after_indexer

Tested with Root and Sub Categories

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-8801](https://jira.corp.magento.com/browse/ENGCOM-8801) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

The result is the same as in the report above https://github.com/magento/magento2/pull/32082#issuecomment-783220455

**Review by danielrussob — APPROVED** [review]

**Pull request merged**